core/domain/learner_playlist_services_test.py

```diff
@@ -100,162 +101,181 @@ def _get_all_learner_playlist_collection_ids(self, user_id):
             learner_playlist_model.collection_ids if
             learner_playlist_model else [])
 
-    def test_mark_exploration_to_be_played_later(self):
+    def test_subscribed_exploration_cannot_be_added_to_playlist(self):
+        # Subscribe to exploration.
+        subscription_services.subscribe_to_exploration(
+            self.user_id, self.EXP_ID_0)
+
+        # Now if we try to add the same exploration, it shouldn't be added.
+        learner_progress_services.add_exp_to_learner_playlist(
```

Comment:
Sorry, I forgot to mention something earlier. Before you perform a "key action", do a check to assert the "before" state. For example, before this call, assert that self._get_all_learner_playlist_exp_ids(self.user_id) returns []. (Then do the action and check the "after" state.)

Ditto for all the tests below.

Reply:
Got it.